rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+
+Inclusions for all AMIs:
+  * [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/)
+    for capturing monitoring data
+  * [InfluxDB](https://www.influxdata.com/) as a data store for Telegraf
+  * [Grafana](https://grafana.com/) to display time series data
+  * [nginx](https://nginx.com) to serve Grafana's front end  
+  * EFF [`certbot`](https://https://certbot.eff.org/) for ACME Certificate
+    Rotation (used by optional ssm `systemd` units)
+  * Python 3 and `pip` for installing and running `certbot`
+  * `uuid` for token generation (used by optional ssm `systemd` units)
+  * `gcc` for ???
+  * `libffi-dev` for ???
+  * `openssl-devel` for ???
+  * `libfontconfig` for ???
+
+Whether or not the actual services are *enabled* depends on post-build 
+actions by the eventual AMI consumer, such as enabling various `systemd`
+units.
+
+### Current Benchmarking
+
+We currently do not test our images against a benchmark.
+
+### Proposed Solution
+
+ * Build a Teleport maintained STIG-high AL2 "Golden" base image using the
+   AWS EC2 Image Builder
+ * Use this hardened image as the base for all Teleport supplied AMIs built
+   with Packer
+ * [Optional] Slimming the AMIs to reduce attack surface
+ * Post-build validation using Amazon Inspector
+ * Ongoing vulnerability scanning with Trivy
+
+#### **CIS vs STIG**
+
+The two best-known standards for security hardening are 
+ * CIS Benchmarks (Level 1 & 2), and
+ * The US DoD Security Technical Implementation Guides (STIGs)
+
+Both of these standards are pretty comprehensive and have wide
+application, although private organisations tend to favour CIS
+over STIG. 
+
+So why choose STIG over CIS?
+
+The decision largely comes down to what tooling is immediately available. Both
+standards are comprehensive - the AL2 CIS benchmark document runs to about 600 
+pages. At an estimated 2 pages per item, that's 300 separate items that need
+detection, remediation and validation. I don't believe that Teleport wants 
+to maintain scripts to do that ourselves, so I have been focused on finding 
+tools to do it for us.
+
+Somewhat unsurprisingly, nearly all of the automated hardening tools I have 
+found are tied to the AWS EC2 Image Builder Tool. Both standards are represented, 
+but while the STIG-compliance components are free for use on any image, the 
+CIS tooling has several drawbacks:
+
+  1. The hardening components require the use of a CIS-supplied base image that
+     incurs an added per-minute royalty to CIS. As we are using this image as a
+     delivery channel to our customers, this is a non-starter.
+  2. The use of the CIS hardening tools requires a subscription to/partnership with 
+     CIS that we do not currently have
+
+The decision to go with STIG is based on being able to improve the situation
+for our customers *now*, while still being able to add CIS hardening tools as
+we develop that relationship in the future.
+
+#### **Hardened Base Image**
```

Comment:
One thing that came to mind, can you please include a section on what STIG hardening _actually is_?